File: ProjectEuler/1-10/project-euler-8.java

Description: Find the greatest K 
consecutive digits in the N digit number.
Completed using Java 8. 
*/ 

import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {
    public static void getMax(int n, int k, String num)
    {
        int product = 1; int max = 0;
        for(int i = 0; i < n - k; i++)
        {
            for(int j = 0; j < k; j++)
            {
                char character = num.charAt(i+j); 
                int m = Integer.parseInt(String.valueOf(character)); 
                product = product * m; 
            }
            if(product > max) max = product; 
            product = 1; 
        }
        System.out.println(max); 
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int t = in.nextInt();
        for(int a0 = 0; a0 < t; a0++){
            int n = in.nextInt();
            int k = in.nextInt();
            String num = in.next();
            getMax(n, k, num); 
        }
    }
}

File: ProjectEuler/project-euler-8.py
""" 
Description: Find the greatest K 
consecutive digits in the N digit number.
Completed using python 3.6.5 
""" 
#!/bin/python3
import sys
import functools
from functools import reduce 

def largest_product(num, n, k):
    max = 0 
    for i in range(n - k):
        p = reduce(lambda x, y: x*y, map(int, num[i:i+k])) 
        max = p if p > max else max 
    print(max)
    return max 

t = int(input().strip())
for a0 in range(t):
    n,k = input().strip().split(' ')
    n,k = [int(n),int(k)]
    num = input().strip()
    largest_product(str(num), n, k)
